packages/valory/customs/uniswap_pools_search/uniswap_pools_search.py

```diff
@@ -0,0 +1,157 @@
+from gql.transport.requests import RequestsHTTPTransport
+
+# Constants
+UNISWAP = "UniswapV3"
+REQUIRED_FIELDS = ("chains", "apr_threshold", "graphql_endpoints", "current_pool")
+
+def check_missing_fields(kwargs: Dict[str, Any]) -> List[str]:
+    """Check for missing fields and return them, if any."""
+    missing = []
+    for field in REQUIRED_FIELDS:
+        if kwargs.get(field, None) is None:
+            missing.append(field)
+    return missing
+
+def remove_irrelevant_fields(kwargs: Dict[str, Any]) -> Dict[str, Any]:
+    """Remove the irrelevant fields from the given kwargs."""
+    return {key: value for key, value in kwargs.items() if key in REQUIRED_FIELDS}
+
+def run_query(query: str, graphql_endpoint: str, variables: Dict = None) -> Dict[str, Any]:
+    """Execute a GraphQL query and return the results."""
+    transport = RequestsHTTPTransport(url=graphql_endpoint)
+    client = Client(transport=transport, fetch_schema_from_transport=True)
+    
+    try:
+        query = gql(query)
+        result = client.execute(query, variable_values=variables)
+        return result
+    except Exception as e:
+        return {"error": f"GraphQL query failed: {str(e)}"}
+
+def calculate_apr(daily_volume: float, tvl: float, fee_rate: float) -> float:
+    """Calculate APR using the formula: (Daily Volume / TVL) × Fee Rate × 365 × 100"""
+    if tvl == 0:
+        return 0
+    return (daily_volume / tvl) * fee_rate * 365 * 100
+
+def get_pools_for_chain(chain: str, graphql_endpoint: str, current_pool: str, apr_threshold: float) -> List[Dict[str, Any]]:
+    """Get all qualifying pools for a specific chain."""
+    graphql_query = """
+    {
+        pools(
+            first: 1000,
+            orderBy: totalValueLockedUSD,
+            orderDirection: desc,
+            subgraphError: allow
+        ) {
+            id
+            feeTier
+            liquidity
+            volumeUSD
+            totalValueLockedUSD
+            token0 {
+                id
+                symbol
+                decimals
+                derivedETH
+            }
+            token1 {
+                id
+                symbol
+                decimals
+                derivedETH
+            }
+        }
+        bundles(where: {id: "1"}) {
+            ethPriceUSD
+        }
+    }
+    """
+    
+    # Execute the GraphQL query
+    data = run_query(graphql_query, graphql_endpoint)
+    if "error" in data:
+        return []
+    
+    qualifying_pools = []
+    pools = data.get("pools", [])
+    
+    for pool in pools:
+        if pool['id'] == current_pool:
+            continue
+            
+        # Calculate pool metrics
+        fee_rate = float(pool['feeTier']) / 1000000  # Convert basis points to decimal
+        tvl = float(pool['totalValueLockedUSD'])
+        daily_volume = float(pool['volumeUSD'])
+        
+        # Calculate APR
+        apr = calculate_apr(daily_volume, tvl, fee_rate)
+        
+        if apr > apr_threshold:
```

Comment:
Add a check to ensure that the selected pool is not the same as the current pool where liquidity has already been invested